test/test_ts_client.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) The Libre Solar Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "tests.h"
+#include <ts_client.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unity.h>
+
+
+void parse_uri_no_subnodes(void)
+{
+    const char *uri = "someID/info";
+    TSUriElems params;
+    params.ts_payload = "";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(1, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING("info", params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING("someID", params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_payload);
+
+}
+
+void parse_uri_with_subnodes(void)
+{
+    const char *uri = "someID/info/";
+    TSUriElems params;
+    params.ts_payload = "";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(0, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING("info/", params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING("someID", params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_payload);
+}
+
+void parse_uri_id_only(void)
+{
+    const char *uri = "someID";
+    TSUriElems params;
+    params.ts_payload = "";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(1, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING("someID", params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_payload);
+}
+
+void parse_uri_empty(void)
+{
+    const char *uri = "";
+    TSUriElems params;
+    params.ts_payload = "";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(-1, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING(NULL, params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING(NULL, params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_payload);
+}
+
+void parse_uri_null(void)
+{
+    const char *uri = NULL;
+    TSUriElems params;
+    params.ts_payload = "";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(-1, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING(NULL, params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING(NULL, params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("", params.ts_payload);
+}
+
+void parse_uri_with_payload(void)
+{
+    const char *uri = "someID/output";
+    TSUriElems params;
+    params.ts_payload = "{Bat_V}";
+    ts_parse_uri(uri, &params);
+    TEST_ASSERT_EQUAL_INT(1, params.ts_list_subnodes);
+    TEST_ASSERT_EQUAL_STRING("output", params.ts_target_node);
+    TEST_ASSERT_EQUAL_STRING("someID", params.ts_device_id);
+    TEST_ASSERT_EQUAL_STRING("{Bat_V}", params.ts_payload);
```

Comment:
Does this make sense? `ts_parse_uri` doesn't touch `ts_payload` at all, so it's obvious that this test passes.

Reply:
Well yes because it should never alter the payload :) The test itself is more of a "signal switch" if anyone ever touches it

Reply:
Ok, then let's keep it. But it's also not valid JSON (irrelevant for this test) ;)